<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>869</width>
    <height>526</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>400</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>1</x>
      <y>1</y>
      <width>345</width>
      <height>26</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="m_layoutTop" stretch="2,0">
     <item>
      <layout class="QHBoxLayout" name="m_connectButtonLayout">
       <item>
        <widget class="QPushButton" name="m_connectButton">
         <property name="text">
          <string>Connect</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="m_disconnectButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Disconnect</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="m_statusLayout" stretch="0,3,0,1,0,0">
       <item>
        <widget class="Line" name="m_vline0">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="m_tuneRevNumberLabel">
         <property name="text">
          <string>Tune:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="m_vline1">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="m_commsLedLabel">
         <property name="text">
          <string>Communications:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLedIndicator" name="m_commsGoodLed" native="true"/>
       </item>
       <item>
        <widget class="QLedIndicator" name="m_commsBadLed" native="true"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>391</width>
      <height>341</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="m_gaugeLayout">
     <item row="0" column="0">
      <widget class="ManoMeter" name="m_revCounter" native="true"/>
     </item>
     <item row="0" column="1">
      <widget class="ManoMeter" name="m_mapGauge" native="true"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="m_revCounterLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Engine Speed</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="m_mapGaugeLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Manifold Pressure</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="ManoMeter" name="m_waterTempGauge" native="true"/>
     </item>
     <item row="2" column="1">
      <widget class="ManoMeter" name="m_airTempGauge" native="true"/>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="m_waterTempLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Engine Temperature</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="m_fuelTempLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="text">
        <string>Intake Air Temperature</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>290</y>
      <width>251</width>
      <height>191</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="m_faultCodesLayout">
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="m_faultCodesLabel">
       <property name="text">
        <string>Fault codes:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLedIndicator" name="m_faultLedCTS" native="true"/>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="m_faultCode01Label">
       <property name="text">
        <string>(01) Coolant temp sensor</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLedIndicator" name="m_faultLedATS" native="true"/>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="m_faultCode02Label">
       <property name="text">
        <string>(02) Air temp sensor</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLedIndicator" name="m_faultLedFuelPump" native="true"/>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="m_faultCode10Label">
       <property name="text">
        <string>(10) Fuel pump ciruit</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLedIndicator" name="m_faultLedTps" native="true"/>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="m_faultCode16Label">
       <property name="text">
        <string>(16) Throttle position sensor</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QLedIndicator" name="m_faultLedTpsSupply" native="true"/>
     </item>
     <item row="5" column="1">
      <widget class="QLabel" name="m_faultCode17Label">
       <property name="text">
        <string>(17) Throttle position sensor supply</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLedIndicator" name="m_faultLedO2Heater" native="true"/>
     </item>
     <item row="6" column="1">
      <widget class="QLabel" name="m_faultCode19Label">
       <property name="text">
        <string>(19) Oxygen sensor heater</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="2">
      <widget class="QPushButton" name="m_clearFaultsButton">
       <property name="text">
        <string>Clear fault codes</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>50</y>
      <width>222</width>
      <height>113</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="m_miscDataLayout" rowstretch="0,0,0,0,0" columnstretch="0,0,1,0">
     <item row="0" column="1" colspan="3">
      <widget class="QProgressBar" name="m_throttleBar">
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="0" column="0">
      <widget class="QLabel" name="m_throttleLabel">
       <property name="text">
        <string>Throttle position:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="1" colspan="3">
      <widget class="QProgressBar" name="m_idleBypassPosBar">
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="m_idleBypassLabel">
       <property name="text">
        <string>Idle bypass position</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLedIndicator" name="m_idleModeLed" native="true"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="m_targetIdleLabel">
       <property name="text">
        <string>Idle switch:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="3" column="1" colspan="2">
      <widget class="QLabel" name="m_gear">
       <property name="text">
        <string>(no reading)</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLabel" name="m_voltage">
       <property name="text">
        <string>0.0V</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="m_gearLabel">
       <property name="text">
        <string>Selected gear:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="m_voltageLabel">
       <property name="text">
        <string>Main voltage:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>400</y>
      <width>373</width>
      <height>87</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="m_loggingLayout">
     <item row="1" column="2">
      <widget class="QPushButton" name="m_startLoggingButton">
       <property name="text">
        <string>Start logging</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2" colspan="2">
      <widget class="QLineEdit" name="m_logFileNameBox"/>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="m_logFileNameLabel">
       <property name="text">
        <string>Log file name:</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QPushButton" name="m_stopLoggingButton">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Stop logging</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>670</x>
      <y>281</y>
      <width>178</width>
      <height>207</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="m_actuatorLayout">
     <item row="0" column="0">
      <widget class="QSlider" name="m_iacPositionSlider">
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="invertedAppearance">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLabel" name="m_iacSetLabel">
       <property name="text">
        <string>0%</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="Line" name="m_hline1">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="2">
      <widget class="QPushButton" name="m_testPTCRelayButton">
       <property name="text">
        <string>PTC (manifold heater) relay</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="QLabel" name="m_testActuatorLabel">
       <property name="text">
        <string>Test actuators:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="2">
      <widget class="QPushButton" name="m_testIdleBypassButton">
       <property name="text">
        <string>Move idle bypass motor</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="2">
      <widget class="QPushButton" name="m_testACRelayButton">
       <property name="text">
        <string>A/C relay</string>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="2">
      <widget class="QPushButton" name="m_testFuelPumpRelayButton">
       <property name="text">
        <string>Fuel pump relay</string>
       </property>
      </widget>
     </item>
     <item row="7" column="0" colspan="2">
      <widget class="QPushButton" name="m_testFuelInjectorButton">
       <property name="text">
        <string>Fuel injector(s)</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0" colspan="2">
      <widget class="QPushButton" name="m_testIgnitionCoilButton">
       <property name="text">
        <string>Ignition coil</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>869</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="m_fileMenu">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="separator"/>
    <addaction name="m_exitAction"/>
   </widget>
   <widget class="QMenu" name="m_optionsMenu">
    <property name="title">
     <string>&amp;Options</string>
    </property>
    <addaction name="m_editSettingsAction"/>
   </widget>
   <widget class="QMenu" name="m_helpMenu">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="m_helpContentsAction"/>
    <addaction name="m_helpAboutAction"/>
   </widget>
   <addaction name="m_fileMenu"/>
   <addaction name="m_optionsMenu"/>
   <addaction name="m_helpMenu"/>
  </widget>
  <action name="m_helpContentsAction">
   <property name="text">
    <string>&amp;Contents...</string>
   </property>
  </action>
  <action name="m_helpAboutAction">
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
  <action name="m_showFaultCodesAction">
   <property name="text">
    <string>Show fault &amp;codes...</string>
   </property>
  </action>
  <action name="m_idleAirControlAction">
   <property name="text">
    <string>&amp;Idle air control...</string>
   </property>
  </action>
  <action name="m_editSettingsAction">
   <property name="text">
    <string>&amp;Edit settings...</string>
   </property>
  </action>
  <action name="m_saveROMImageAction">
   <property name="text">
    <string>&amp;Save ROM image...</string>
   </property>
  </action>
  <action name="m_exitAction">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ManoMeter</class>
   <extends>QWidget</extends>
   <header>manometer.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QLedIndicator</class>
   <extends>QWidget</extends>
   <header>qledindicator.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
